backend/app/services/diagnosis_agent.py

```
@@ -32,462 +37,552 @@ class DiagnosisState(TypedDict):
+    return {"weather_data": {"status": "에러", "message": last_error}}
 
-## ⚠️ 공지: 제공된 정보는 공공데이터에 기반한 참고용입니다. 농약 사용 전 반드시 제품 라벨의 규정을 확인하십시오.
---------------------------------------------------
-"""),
-        ("user", "데이터를 기반으로 템플릿 엔진의 역할을 수행하라.")
-    ])
+from app.models.ncpms import NcpmsDiagnosis
+
+async def fetch_ncpms(state: DiagnosisState) -> dict:
+    pest, crop = state.get("pest", "알 수 없음"), state.get("crop", "알 수 없음")
+    if not pest or pest == "알수없음": return {"ncpms_data": "해충명 정보가 없습니다."}
```

Ruff (0.15.10)

[error] Multiple statements on one line (colon)

(E701)